internal/cloudapi/v2/compose.go

```diff
@@ -1002,6 +1003,15 @@ func (request *ComposeRequest) GetImageRequests(distroFactory *distrofactory.Fac
 			return nil, err
 		}
 
+		// If no repositories are included with the imageRequest use the defaults for the distro
+		if len(ir.Repositories) == 0 {
+			dr, err := repoRegistry.ReposByImageTypeName(distribution.Name(), arch.Name(), imageType.Name())
```

Comment:
I think that you should use `request.Distribution` instead of `distribution.Name()` here. If the DistroFactory has an alias set for the distro name specified in the ImageRequest, then the `distribution.Name()` would return the distro name of the target distro of the alias.

This is done also in Weldr, see https://github.com/osbuild/osbuild-composer/blob/8f28da5214f4078593dea051d1eee25727ab1473/internal/weldr/api.go#L3568-L3571

Reply:
Are you sure that will work? I see obvious alias code in distrofactory but not in the reporegistry code.

Reply:
I'm pretty sure, since it is tested in Weldr by:
https://github.com/osbuild/osbuild-composer/blob/073e304978acade8bfa059f00a005402aa037c99/test/cases/weldr-distro-dot-notation-and-aliases.sh#L318-L341

The idea is that if one specifies a distro name in the request, we always need to use it as is. That means that we use that string to get the distro instance and repositories.

In DistroFactory, this means first translating the potential alias to the target distro name and getting the Distro object (however, that won't reflect the aliased name on its own).

For repositories, this means grabbing the repos that correspond to the `<distro_name>.json`. If one specified an alias, there must be repositories defined for that alias distro name. If there are no repos, one can't build that distro (which is consistent with e.g. missing el9 repos on el8 - the distro definition is there, but you can't build it because of missing repos). For this reason, there's no aliasing code in repositories, since repo definitions must be always explicit. This is how it worked before DistroFactory and I kept the behavior intentionally.

I admit that it would be good to capture this information somewhere, since it is not obvious. Happy to do so if you have any suggestions. :wink:

Reply:
Ok, I think I understand now. The bit I was missing was that the alias is just an alias for the distro definition, NOT for the repository json file. I was thinking, for example, `rhel-9` would then get aliased to `rhel-9.3` and use the `rhel-9.3.json` repo file. But it doesn't, it uses `rhel-9.json` which points to the latest release on the cdn.